tests/test_testclient.py

```diff
@@ -147,6 +148,72 @@ async def loop_id(request):
     assert first_task is not startup_task
 
 
+class ExceptionGroup(anyio.ExceptionGroup):
+    def __init__(self, exceptions):  # pragma: no cover
+        super().__init__()
+        self.exceptions = exceptions
+
+
+def test_testclient_cancel_propagate_lifespan_failure(test_client_factory):
+    lifespan_taskgroup = None
+    event = None
+
+    @asynccontextmanager
+    async def lifespan(app):
+        nonlocal lifespan_taskgroup, event
+        event = anyio.Event()
+        async with anyio.create_task_group() as lifespan_taskgroup:
+            yield
+
+    app = Starlette(lifespan=lifespan)
+
+    @app.route("/throw")
+    async def throw(self):
+        async def throw():
+            await event.wait()
+            raise MyError
+
+        lifespan_taskgroup.start_soon(throw)
+        return JSONResponse("good")
+
+    @app.route("/ping")
+    async def ping(self):
+        return JSONResponse("pong")
+
+    @app.route("/sleep_forever")
+    async def sleep_forever(self):
+        event.set()
+        await anyio.sleep_forever()
+
+    class MyError(Exception):
+        pass
+
+    client = test_client_factory(app)
+
+    client.get("/ping").json() == "pong"  # app works outside of lifespan
+
+    for _ in range(2):  # context manager can be re-used
+        try:
+            client.__enter__()
+            assert client.get("/ping").json() == "pong"
+            assert client.get("/throw").json() == "good"
+            with pytest.raises(CancelledError):
+                client.get("/sleep_forever")
+            with pytest.raises(ClosedError):
+                client.get("/sleep_forever")
+        except BaseException as e:  # pragma: no cover
+            try:
+                client.__exit__(*sys.exc_info())
+            except BaseException as f:
+                raise ExceptionGroup((e, f)) from None
```

Comment:
I want to move this ExceptionGroup into testclient but constructing it isn't public